exercises/affine-cipher/canonical-data.json

```diff
@@ -0,0 +1,191 @@
+    "* Encoding from English to affine cipher",
+    "* Decoding from affine cipher to all-lowercase-mashed-together English"
+  ],
+  "cases": [
+    {
+      "description": "encode",
+      "comments": [ "Test encoding from English to ciphertext with keys" ],
+      "cases": [
+        {
+          "description": "encode yes",
+          "property": "encode",
+          "input": {
+            "phrase": "yes",
+            "key": {
+              "a": 5,
+              "b": 7
+            }
+          },
+          "expected": "xbt"
+        },
+        {
+          "description": "encode no",
+          "property": "encode",
+          "input": {
+            "phrase": "no",
+            "key": {
+              "a": 15,
+              "b": 18
+            }
+          },
+          "expected": "fu"
+        },
+        {
+          "description": "encode OMG",
+          "property": "encode",
+          "input": {
+            "phrase": "OMG",
+            "key": {
+              "a": 21,
+              "b": 3
+            }
+          },
+          "expected": "lvz"
+        },
+        {
+          "description": "encode O M G",
+          "property": "encode",
+          "input": {
+            "phrase": "O M G",
+            "key": {
+              "a": 25,
+              "b": 47
+            }
+          },
+          "expected": "hjp"
+        },
+        {
+          "description": "encode mindblowingly",
+          "property": "encode",
+          "input": {
+            "phrase": "mindblowingly",
+            "key": {
+              "a": 11,
+              "b": 15
+            }
+          },
+          "expected": "rzcwa gnxzc dgt"
+        },
+        {
+          "description": "encode numbers",
+          "property": "encode",
+          "input": {
+            "phrase": "Testing,1 2 3, testing.",
+            "key": {
+              "a": 3,
+              "b": 4
+            }
+          },
+          "expected": "jqgjc rw123 jqgjc rw"
+        },
+        {
+          "description": "encode deep thought",
+          "property": "encode",
+          "input": {
+            "phrase": "Truth is fiction.",
+            "key": {
+              "a": 5,
+              "b": 17
+            }
+          },
+          "expected": "iynia fdqfb ifje"
+        },
+        {
+          "description": "encode all the letters",
+          "property": "encode",
+          "input": {
+            "phrase": "The quick brown fox jumps over the lazy dog.",
+            "key": {
+              "a": 17,
+              "b": 33
+            }
+          },
+          "expected": "swxtj npvyk lruol iejdc blaxk swxmh qzglf"
+        },
+        {
+          "description": "encode with a not coprime to m",
+          "property": "encode",
+          "input": {
+            "phrase": "This is a test.",
+            "key": {
+              "a": 6,
+              "b": 17
+            }
+         },
+          "expected": "Error: a and m must be coprime."
```

Comment:
If the keys are `a` and `m`, why are they listed as `a` and `b` in the canonical data?

Reply:
`m` is the length of the alphabet. `a` and `b` make the key for encryption and decryption.
Since the description of the problem has it working with English, `m` will always be 26.

Reply:
Also, `a` and `m` must be coprime but `b` can be any integer value.

Reply:
Makes sense; that's my error.